### PR TITLE
RA Balance changes for March 2020

### PR DESCRIPTION
**Pull request by PunkPun**
## Briefing
These changes have been in the works for a while, they have been playtested and iterated for months. One of the changes hasn't but is a must, removal of opportunity fire from aircraft

### Changing all small-caliber weapons to instant hit
- Bullets take time to travel. The farther a target is, the less damage moving targets receive. As it stands most small caliber weapon ranges are small and unit movement speed is slow enough that the majority of units take full damage regardless of this complex system. This change makes all targets receive full damage regardless of movement and range. It mostly impacts Pillboxes and Mobile Flak anti-air capabilities. Bullets having speed is not mentioned anywhere in the game and very few people are even aware of it. My goal is to remove this unnecessary complexity
- Pillboxes are special because they shoot 6 bursts, but unlike all other assets in the game, they do not retarget in between them. It means that moving targets will still take less damage than stationary ones. The only difference is that damage will no longer depend on distance from the Pillbox to the target nor if the target is on roads. Rangers instead of taking no damage at maximum range, will take a little. Dogs instead of surviving the first pillbox volley, will die. That is about it
- Effect to Mobile Flak anti-air is by far the most significant. In summary Yaks and MiGs are fast, always moving and Mobile Flak has long anti-air range. All of this summed up means that MiGs do not even take damage at Flaks maximum range. Combined with SAM Site having very little effective range against MiGs makes it hard for Soviets to kill them. Whether they die mostly depends on the attentiveness of the person controlling them and not the efforts of the opponent to counter play. With instant-hit Flaks will now be able to properly counter. To substitute for this MiGs are getting a health buff (there is more info on it). This change is also more justifiable considering that AA Guns bullets had already been made faster a long time ago meaning they already have been dealing full damage

### Yak
- Loses opportunity fire
   - Currently Yaks are overperforming. Although the way Yak works is fun, it is really hard to balance them. They can not be at the same time a good sniper unit and an infantry mass murderer. As Yaks are planned to be reworked in the next +1 release, we decided not to try and mix those two roles together and just revert the change

### Longbow
- Loses opportunity fire
   - So it would be consistent between all aircraft. Opportunity fire does not really affect Longbow in meaningful ways
- Reload delay from 60 to 30
- Anti-air missiles
   - Speed from 384 to 492
   - Horizontal rate of turn from 10 to 25
   - Near enough from 0c298 to 0c596
   - Range from 5 to 4 cells
      - Longbows will no longer miss planes, once fired missiles instead of sometimes exploding due to bad tracking will always connect, they already did versus helicopters 
- Anti-ground missiles
   - Burst amount from 4 to 2
   - Burst delay from 14 to 7
      - Missiles will be burstier and will overkill less in big groups. Total ammo empties by about 1 second faster. This makes burstiness the same between air / ground weapons. Burst delay is done in a way that if Longbow fired at stationary target no ammo would be wasted

### MiG
- Health from 7500 to 10000
   - MiGs will take a lot more damage from Flaks because of the insta-hit changes, additionally Longbow gets an upgrade vs planes. It is a counter buff, additionally the damage from AA Guns make MiGs a very risky unit in SvA matchup. A health buff will make them more viable
   - It will be killed by 5 Rocket Soldier rockets instead of 4
- Loses opportunity fire
   - So it would be consistent between aircraft. Opportunity fire does not really affect MiG in meaningful ways

### Black Hawk 
- Loses opportunity fire
   - So it would be consistent between aircraft. Opportunity fire does not really affect BH in meaningful ways

### Chinook
Health from 14000 to 20000
   - They have lower vision than AA Guns have weapon range. This change increases the much needed time for reaction
   - As a consequence It can barely land 4 cells in front of a vision supported AA Gun for a one way trip. If you go deeper Chinook dies

### Chrono Tank
- Rocket damage made the same as of Rocket Soldier
   - Damage from 2500 to 5000
   - Versus
      - none from 28% to 10%
      - wood from 88% to 74%
      - light from 88% to 34%
      - heavy from 120% to 100%
      - concrete from 60% to 50%
- Movement speed from 113 to 96
   - Medium Tank has Speed 82, Mobile Flak & Light Tank 118
      - Its counters will relatively become much faster than it. The portable chono ability will be the only way to outmaneuver its hunters. CTanks will have a harder time escaping after blinking aggressively 
- Health from 45000 to 40000
   - Instead of needing a second aircraft, you will need to use just 1 full ammo clip to kill it. This change was supposed to go with the Tesla Tank health nerf, but none of the Chrono Tank changes were implemented last release, they still lacked testing. This time with some minor adjustments we are more confident

Currently Chrono Tank has no role to fill, everything they do Allies can do better and more reliably with other units. With these changes Chrono Tank becomes the best ground anti-armour unit. It deals incredible damage to heavy armour, can chase anything with the Portable Chrono ability, it receives reduced damage from most of the heavy armoured units as it has light armour itself, and a decent sized health pool.

Apart from aircraft Mobile Flaks are the best Soviet hard counter, although 1 Flak (700) barely loses to 1 Chrono Tank (1350) as the numbers of units rise the better Flaks become. They are much less bursty and can shoot while moving / turning giving them an edge in those engagements

Apart from aircraft Allies can counter with Light Tanks and Rangers, they do not trade as well as Flaks but still do it cost efficiently. It is important to focus fire CTanks with them. Additionally Allies have MRJs which can completely nullify CTank rockets

### Phase Transport
- Carry size from 4 to 5
   - To be more effective at dropping, it will not impact the game significantly
- Same damage buffs / nerfs as Chrono Tank as they both share the same weapon
- Splitting weapon from CTank and increasing reload delay from 60 to 100
   - In tests they were proven to be able to kill tanks cost efficiently in larger numbers. It is not the role this unit is supposed to have. 
   - Reload delay will be twice as much as Rocket Soldiers

### Minelayer 
- Will not receive damage from mines
   - When a Minelayer is caught laying mines it can be insta-killed as any shots to it will also damage mines beneath it. As of this change mines will explode but they will not kill the Minelayer. Current behaviour seems like a bug rather than a feature

### CODE REVIEW

File: mods/ra/weapons/missiles.yaml [hunk 1]
```diff
@@ -212,13 +215,17 @@ APTusk:
...
-			Wood: 88
-			Light: 88
-			Heavy: 120
-			Concrete: 60
+			None: 10
+			Wood: 74
+			Light: 34
+			Heavy: 100
+			Concrete: 50
+
```
Comment (abcdefg30) [inline mods/ra/weapons/missiles.yaml]: This should be removed.

**Comment by pchote** [comment]
The soviet campaign missions will probably need to be updated to account for the chinook.

File: mods/ra/weapons/missiles.yaml (hunk 1 above)
Comment (PunkPun) [reply mods/ra/weapons/missiles.yaml]: Removed

Comment (PunkPun) [reply mods/ra/weapons/missiles.yaml]: Removed the tabs as well

**Comment by PunkPun** [comment]
removed `ScanOnIdle: false` from planes

**Comment by PunkPun** [comment]
From my investigation only "OpenRA Originals" missions even have helipads, the 3 bottom ones. And afaik in the OG chinook was only buildable in multiplayer. Should I just make chinook unbuildable in all soviet missions?

**Comment by pchote** [comment]
Lets start with that, yes.

**Comment by PunkPun** [comment]
Done that

**Comment by abcdefg30** [comment]
As the topic came up on IRC: I see this is titled "changes for March 2020". I'm sorry, but this PR was filed too late for this release. It should have been filed *before* the first playtest to allow for preceding discussion, and to allow testing and more discussion between first and the usual second playtest. We're in a feature freeze now (which makes the potentially required mission changes more problematic). Our modding community doesn't get an exception, and I think it would be unfair to risk the next release by making an exception for the RA competitive scene.

**Comment by netnazgul** [comment]
At least the Yak change should be split and merged then, it doesn't break missions and actually is even better for missions because current behaviour wasn't balanced around at all.

Artillery and Turret tweaks might also feature now and not be shelved for later as they don't have missions impact. As is Minelayer fix.

**Comment by PunkPun** [comment]
All of these changes are relatively low risk, have already been tested and debated. It's very unlikely that we'd need to change them in the next playtest. I can shaft the chinook one a second time as it is the only one that can break missions. Others should not effect in them in significant ways. I was under the impression that we always merge balance changes after the first playtest, not that I had much time to run tests before it.

**Comment by PunkPun** [comment]
I removed Longbow damage buff, turret and arty changes because they are related with the goals for the next balance update. Additionally they are the newest of the bunch and still have some light opposition. I also removed the chinook being shared across factions because it has a high probability of breaking missions and a second playtest isn't planned

**Comment by reaperrr** [comment]
Needs a rebase.

**Comment by PunkPun** [comment]
rebased

**Review by pchote — APPROVED** [review]

File: mods/ra/weapons/smallcaliber.yaml [hunk 2]
```diff
@@ -272,8 +268,8 @@ M60mg:
 	Report: gun5.aud
 	ValidTargets: Ground, Infantry
 	InvalidTargets: Vehicle, Water, Structure, Wall, Husk, Mine
-	Projectile: Bullet
-		Speed: 1c682
+	Projectile: InstantHit
+		Blockable: true
```
Comment (pchote) [inline mods/ra/weapons/smallcaliber.yaml:272]: Is this change intentional? I don't think you explained it in the description.

**Review by abcdefg30 — APPROVED** [review]

File: mods/ra/weapons/smallcaliber.yaml (hunk 2 above)
Comment (pchote) [reply mods/ra/weapons/smallcaliber.yaml:272]: Ok, I see. `Bullet` defaults to `Blockable: true` and `InstantHit` to `Blockable: false`.

Comment (PunkPun) [reply mods/ra/weapons/smallcaliber.yaml:272]: Yeah, it's strange in the way defaults are set up. But kinda makes sense in the way ORA mods work

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/1d5980981adb053f69e3aa67a771a1f1574e06bc...9a251c5375bbdcd909cfb0fc9ca9d05a31a1c813)

https://github.com/OpenRA/OpenRA/commit/00ddf82e8509cd90752dcc90926b61b2472316f8
